draft-ietf-tls-tls13.md

```diff
@@ -2785,6 +2792,22 @@ certificate_authorities
   the certificate_authorities list is empty, then the client MAY
   send any certificate of the appropriate ClientCertificateType,
   unless there is some external arrangement to the contrary.
+
+certificate_extensions
+: A list of certificate extension OIDs [RFC5280] with their allowed
+  values, represented in DER-encoded format. Some certificate
+  extension OIDs allow multiple values (e.g. Extended Key Usage).
+  If the server has included a non-empty certificate_extensions list,
+  the client certificate MUST contain all of the specified extension
+  OIDs that the client recognizes. For each extension OID recognized
+  by the client, all of the specified values MUST be present in the
+  client certificate. However, the client MUST ignore and skip any
+  unrecognized certificate extension OIDs. If the client has ignored
+  some of the required certificate extension OIDs, and supplied a
+  certificate that does not satisfy the request, the server MAY at
+  its discretion either continue the handshake without client
+  authentication, or respond with a fatal unsupported_certificate
```

Comment:
Note: need to validate that either (a) certificates never contain duplicate extensions or (b) you can have multiple instances of the same extension.